internal/dispatch/stream.go

```diff
@@ -0,0 +1,115 @@
+package dispatch
+
+import (
+	"context"
+	"sync"
+
+	grpc "google.golang.org/grpc"
+)
+
+// Stream defines the interface generically matching a streaming dispatch response.
+type Stream[T any] interface {
+	// Publish publishes the result to the stream.
+	Publish(T) error
```

Comment:
Why not `Send` here too to match grpc / channel naming?

Reply:
I *explicitly* named it differently to prevent sending a gRPC stream (which is not concurrent safe) into this interface (which all callers expect to be safe to use concurrently)